projects/components/src/label-tag/label-tag.component.ts

```diff
@@ -0,0 +1,23 @@
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+import { Color } from '@hypertrace/common';
+
+@Component({
+  selector: 'ht-label-tag',
+  styleUrls: ['./label-tag.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="label-tag" [ngStyle]="{ backgroundColor: this.backgroundColor, color: this.labelColor }">
+      {{ this.label }}
+    </div>
+  `
+})
+export class LabelTagComponent {
+  @Input()
+  public label?: string;
+
+  @Input()
+  public labelColor?: Color;
+
+  @Input()
+  public backgroundColor?: Color;
```

Comment:
No sure about making these styles as inputs. Can we do it by introducing an input for roles or style enum  and hiding all the required css changes inside the component? An example of this is the button component. 

It would improve readability as we would not have to go into the details of choosing the right color and background. We can just choose a role/style and all corresponding css would come. 

It would also allow us to not maintain a color js object.

Reply:
Any particular reason why we don't want to maintain a color js object? I feel its good given we use so many hex strings in the code already.

Also, for style inputs, I think this works best for the tag use case because we might want to use it for a varied set of usecases. Example, `Beta` / `New` or even other usecases in the future. I feel that the role/style using CSS will be cumbersome to maintain and won't give flexibility for a very generic component like this.

Reply:
Regarding color object, there was some discussion before about doing this with a service that would read directly from css variables. The advantages are
1. we wouldn't have to maintain colors at two places. With the current color object approach, if Michael changes any of the color (as part of rebranding), we would have to change at both places. 
2. As this color object grows, it would become difficult to maintain. 
3. Implementing Themeing would be a big problem. For dark theme, we wouldn't be applying the same colors. 

The service approach would work best as it will only read from one source of truth and can quickly switch from one theme to another.

Similarly, by exposing label colors and background colors as input properties, we are :
1. In a way sacrificing on the component abstraction for more flexibility. 
2. If Michael choose a different style, we will have to modify all its usages to pass new colors, rather than modifying the css from within this component.
3. Themeing becomes way more harder as every component usage is passing colors explicitly. If this pattern grows, it will become much more difficult. 


If we just introduce roles, we can achieve the same thing with all the flexibility, just that we are pushing it one layer deep.

Reply:
I don't love the idea of maintaining two separate color declarations, but I think it's strictly better than the hardcoding we're doing now - maybe we should go with it and revisit when we have more time to think on color in general. The idea of roles is independent vs explicit values is independent of this, and I generally do agree there - the problem being that we don't want to have the generic component understand where it's being used (if it's supplying the roles, which in a tag component can be pretty varied and specific).

So I guess what I'm saying is - without knowing what we're doing with theming (and now's not the time for that) it's hard to say which solution will be easier to adapt in the future, and I'd personally be fine with either way right now.

Reply:
That was what I was aiming for also. 2 sources is better than all the string values we have lying around the code right now. I was planning on cleaning those up as well. When we revisit theming and define a color service, this work isn't going to hurt us. If anything, it'll make life easier compared to moving from what we have now.

And yes, my reasoning for specific colors also was the same. Tag component can be very varied, I don't think we should push that use case specific styling down to the component.

Like @aaron-steinfeld mentioned in another comment. If we have a use case like beta-tag, which is going to be used at different places, we can actually create another small component on top of this one to make life easier. I feel thats a cleaner solution than pushing styling as roles into this generic component.

@anandtiwary thoughts?

Reply:
Isn't a beta tag component over label tag same as combining them into one component and selecting a role?. That also works, but we are adding another component which is not doing much. 

Do we want to use this pattern elsewhere? Like, what are your thoughts on applying the same pattern on button component? We would have a single button component that would take different css as inputs and separate specific button components like `button-primary`, `button-additive`, `button-destructive` etc. Would you prefer it over a single component?

That being said, it is still a small component, so I am not too worried. We can surely iterate over it. :)

Reply:
@anandtiwary technically, its kinda the same thing. We customize a component for the usecase. If we have just a one off use case, the label component can be directly used with colors as inputs. If there are multiple use cases, we can write a wrapper component (like `beta-tag`) to avoid duplication.

IMO, button usecase is slightly different from this because our design system choices would dictate the kind of buttons we use. We won't use many different styles of buttons nor would we want one off customizations for it. And the styles of a component like button is dictated by the functionality of it. Whereas a tag is just a text with a style (style being the differentiator) and theoretically, it could have any style applied to it.

If this tag ever becomes an actionable tag (ex: click to open something, show a popover etc), then I'd see more merit in defining role based styles. Because then we would want the user to relate a particular style with a particular action also. For a display only tag, I feel it should be left customizable, in control of the consumer.

Reply:
It's the same idea but an inversion of concerns - a button with multiple roles means the base component has knowledge of the whole design system. For a button where we have a limited number based on an abstract concept like primary/additive/destructive, that makes sense. New concepts shouldn't be added with any regularity, they should be driven by the design system, and we want to prevent any stray from the specific predefined roles. On the other hand, the tag's categories aren't really look and feel driven, but use case - such as a beta tag. So building use cases in to the base component there seems like the wrong approach, as these additions may be driven by specific features rather than the design system, and we don't want feature changes to require changes to the underlying component system.

Reply:
Seems like we are on the same page on this.

Reply:
Merged this change for now. We can iterate on this.